src/components/form-field/search-field/search-field.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useMemo, useRef } from "react";
+import { classNames } from "../../../util/class-names";
+import CrossIcon from "../../../icons/cross-icon";
+import SearchIcon from "../../../icons/search-icon";
+
+export interface SearchInputProps extends React.ComponentPropsWithoutRef<"input"> {
+    autoSelect?: boolean;
+    ariaDescribedBy?: string;
+    error?: boolean;
+    onClear: () => void;
+}
+
+export const SearchInput = ({
```

Comment:
My suggestion here would be to match the component name with the filename. The filename says search-field and mismatch our general naming convention. :)